### PR TITLE
Config migrate SAML keys and certificates

### PR DESCRIPTION
**Pull request by RajatVaryani**
#### Summary
Config migrate SAML keys and certificates

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/10654

### CODE REVIEW

**Comment by jasonblais** [comment]
Thanks @RajatVaryani for the contribution! Let us know when it's ready for a review

**Comment by RajatVaryani** [comment]
@jasonblais Can I get quick feedback from someone on the PR?

**Comment by jasonblais** [comment]
@jwilander maybe you can help with an initial review, since the Jira ticket is in the Cloud team? Feel free to reassign review

File: config/migrate.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	}
+
+	sourceConfig := sourceStore.Get()
+	_, err = destinationStore.Set(sourceConfig)
+
+	if err != nil {
+		return errors.Wrapf(err, "failed to set config")
+	}
+
+	idpCertificateFile, err := sourceStore.GetFile(*sourceConfig.SamlSettings.IdpCertificateFile)
```
Comment (jwilander) [inline config/migrate.go]: I'd suggest checking the error here. If the error is only a 'not found' error then continue like you have but if it's an error for some other reason (e.g. permissions) we should fail with that error

**Comment by RajatVaryani** [comment]
@jasonblais It's ready for review.

File: config/migrate.go (hunk 1 above)
Comment (RajatVaryani) [reply config/migrate.go]: I have made the changes

File: config/migrate.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func migrateFile(name string, source Store, destination Store) error {
+	fileExists, err := source.HasFile(name)
+
```
Comment (lieut-data) [inline config/migrate.go]: nit: propose removing empty line here

Comment (RajatVaryani) [reply config/migrate.go]: Done.

**Comment by RajatVaryani** [comment]
Can someone please merge this?

**Comment by jasonblais** [comment]
Thanks @RajatVaryani for the quick changes!

I'm sure devs will merge the changes once they come back from the weekend :)

**Pull request merged**